### PR TITLE
[docs] Add bugfix release criteria

### LINKED ISSUE #5681: Document requirements for patch releases
> We had a relatively bad bug on 0.55.0 which would crash the collector when the telemetry level was set to none. This did not affect the default config. Is this enough for a patch release while we are still in v0.x?

Follow up from July 13 SIG meeting, we need to define the requirements for a patch release.

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:**  Add criteria describing when to make a bugfix release. This is basically a rehash of https://github.com/open-telemetry/opentelemetry-collector/issues/5681#issuecomment-1185283351 with nicer language and ordered from least to most ambiguous.

**Link to tracking Issue:** Fixes #5681

cc @codeboten @jpkrohling

### CODE REVIEW

File: docs/release.md [hunk 1]
```diff
@@ -98,6 +98,31 @@ The last step of the release process creates artifacts for the new version of th
...
+    - The bug is not specific to a platform that is not widely used
+    - The bug happens with the default configuration or with a commonly used one (e.g. has been reported by multiple people)
+5. The bug is sufficiently severe. For example (non-exhaustive list):
+    - The bug makes the Collector crash reliably
+    - The bug makes the Collector fails to start under an accepted configuration
+    - The bug produces significant data loss
+    - The bug makes the Collector negatively affect its environment (e.g. significantly affects its host machine)
+
+The OpenTelemetry Collector maintainers will ultimately have the responsibility to assess if a given bug fulfills all the necessary criteria.
+Exceptions may be granted in a case-by-case basis if there is widespread consensus that a bugfix release should be done despite the above rules.
```
Comment (mx-psi) [inline docs/release.md:124]: Should this line be here?

File: docs/release.md [hunk 2]
```diff
@@ -98,6 +98,31 @@ The last step of the release process creates artifacts for the new version of th
...
+
+When considering making a bugfix release on the `v0.N.x` release cycle, the bug in question needs to fulfill the following criteria:
+
+1. The bug was introduced on the `v0.N-1.x` release cycle.
+2. The bug has been reported within the first 3 working days after the official binaries were released.
+3. The bug has no workaround or the workaround is significantly harder to put in place than updating the version. Examples of simple workarounds are:
+    - Reverting a feature gate.
+    - Changing the configuration to an easy to find value.
+4. The bug happens in common setups. To gauge this, maintainers can consider the following:
+    - The bug is not specific to a platform that is not widely used
```
Comment (TylerHelmuth) [inline docs/release.md:115]:
```suggestion
    - The bug is not specific to an uncommon platform
```
The double "not"s meesed me up lol

File: docs/release.md [hunk 3]
```diff
@@ -98,6 +98,31 @@ The last step of the release process creates artifacts for the new version of th
...
 
+## Bugfix releases
+
+### Bugfix release criteria
+
+Both `opentelemetry-collector` and `opentelemetry-collector-contrib` have very short 2 week release cycles. Because of this, we put a high bar when considering making a patch release, to avoid wasting engineering time unnecessarily.
+
+When considering making a bugfix release on the `v0.N.x` release cycle, the bug in question needs to fulfill the following criteria:
+
+1. The bug was introduced on the `v0.N-1.x` release cycle.
```
Comment (TylerHelmuth) [inline docs/release.md:109]: Does this mean in order to release a `0.55.1` patch, the bug had to appear in `0.55-1.0`, which is `0.54.0`?  Am I interpreting this correctly?

Comment (mx-psi) [reply docs/release.md:109]: Yes, that's it. Do you think I should phrase it differently?

Comment (TylerHelmuth) [reply docs/release.md:109]: I think I was expecting a patch release to deal with issues that appeared in the release itself.  Like if 0.55.0 was released in a state such that the collector wouldn't even start, I was expecting a 0.55.1 release immediately, not waiting for 0.56.0 and then 0.56.1.

Combined with criteria #2, it sounds like in order for an issue from 0.55.0 to get patched, it has to be found on Wednesday/Thursday, Friday, or Saturday after the 0.55.0 release, then wait 11 more days for a 0.56.0 release, and then wait for a 0.56.1 release.

Should the first criteria be "The bug was introduced in version `v0.N-1.x` or higher, up to and including the current release"?

File: docs/release.md (hunk 1 above)
Comment (TylerHelmuth) [reply docs/release.md:124]: I like the wiggle room, but I think ultimately the "case-by-case" basis is based on the maintainer's judgement.  Could modify the line above to be "The OpenTelemetry Collector maintainers will ultimately have the responsibility to assess if a given bug fulfills all the necessary criteria, or should be patched anyways".

File: docs/release.md (hunk 3 above)
Comment (mx-psi) [reply docs/release.md:109]: Right, sorry, I meant `v0.N.0`, I don't know why I wrote that :smile:

> I was expecting a patch release to deal with issues that appeared in the release itself.

This is what I wanted to write, let me rephrase that

Comment (mx-psi) [reply docs/release.md]: Replaced `N-1`by `N`, hopefully that's clear enough

**Review by TylerHelmuth — APPROVED** [review]

File: docs/release.md (hunk 1 above)
Comment (mx-psi) [reply docs/release.md]: Reworded

**Review by jpkrohling — APPROVED** [review]

**Review by codeboten — APPROVED** [review]
